app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java

```diff
@@ -70,15 +70,33 @@ protected void onCreate(Bundle savedInstanceState) {
         }
 
         if (getIntent().hasExtra("inputs")) {
-            System.out.println(getIntent().getStringExtra("inputs").getClass());
-            inputs = new Gson().fromJson(getIntent().getStringExtra("inputs"), StudyParameter[].class);
+            try {
+                String x = getIntent().getStringExtra("inputs");
+                x=x.replace("\"[","[");
```

Comment:
@NachoMoment, something isn't right about all these string replacements. At no other point in this project do we do this for any javascript json response. The correct way would be to find out why the input/output parameter json isn't valid.

Reply:
@jdrichards50 that was my initial approach which lead me to the getStudyOutputPrameters function in ChartIQ.java and this is how the result comes back from the getStudyParameters function, so this is how we are returning it from the library, right? We would need to change how we return it in the library to avoid doing the replacements.

Reply:
Yes, most likely nativeSdkNative is incorrectly returning the study parameters.